src/System.Net.Sockets/tests/FunctionalTests/DisconnectTest.cs

```diff
@@ -0,0 +1,165 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+// TODO: make sure everything's OuterLoop and Windows-only!
+
+using System.Net.Test.Common;
+using System.Threading;
+
+using Xunit;
+using Xunit.Abstractions;
+
+namespace System.Net.Sockets.Tests
+{
+    public class DisconnectTest
+    {
+        private readonly ITestOutputHelper _log;
+
+        public DisconnectTest(ITestOutputHelper output)
+        {
+            _log = TestLogging.GetInstance();
+            Assert.True(Capability.IPv4Support() || Capability.IPv6Support());
+        }
+        public void OnCompleted(object sender, SocketAsyncEventArgs args)
+        {
+            EventWaitHandle handle = (EventWaitHandle)args.UserToken;
+            handle.Set();
+        }
+
+        [Fact]
+        [OuterLoop]
```

Comment:
You should either put a comment on this [OuterLoop] to say why it is outer loop.  Or, better, use the new attribute for [OuterLoop] that specifies a reason that it is outer loop.

See #11345